src/scm/bitbucket.js

```diff
@@ -1,4 +1,277 @@
+    .then(response => {
+      return getAllItems(Promise.resolve({
+        token: this.accessToken,
+        items: [], 
+        urls: [],
+        url: `${this.baseUrl}/repositories/${context.repo.fullName}/src/${response.target.hash}/?access_token=${this.accessToken}`
+      }),
+      this.followDirectory,
+      'bitbucket')
+      .then(response => {
+        const promises = response.map(src => {
+          return new Promise((resolve, reject) => {
+            $.get(src.links.self.href, { access_token: this.accessToken })
+            .then(content => {
+              resolve({ file: src.path, content: content});
+            })
+            .fail(reject)
+          });
+        });
+        return Promise.all(promises);
+      });
+    });
+  }
+
+  getRepos() {
+    return this.getAccessToken()
+    .then(accessToken => {
+      return getAllItems(Promise.resolve({
+        token: accessToken,
+        items: [], 
+        url: `${this.baseUrl}/repositories/?access_token=${accessToken}&q=scm="git"&role=contributor`
+      }),
+      this.followPaginate,
+      'bitbucket');
+    })
+    .then(response => {
+      const repos = response.map(repo => repo.full_name);
+      //if current bind still existed, use it
+      const repo = context.bindRepo[context.id];
+      if (repo && $.inArray(repo.fullName, repos) >= 0) {
+        context.repo = repo;
+      }
+      return repos;
+    });
+  }
+
+  createRepo() {
+    const repo = $('#new-repo-name').val();
+    const desc = $('#new-repo-desc').val();
+    const isPrivate = $('#new-repo-type').val() !== 'public';
+    const payload = {
+      scm: 'git',
+      description : desc,
+      is_private: isPrivate
+    }
+    if (!repo || repo === '') return;
+    this.getAccessToken()
+    .then(() => {
+      return $.ajax({
+        url: `${this.baseUrl}/repositories/${this.user}/${repo}`,
+        headers: {
+          'Authorization': `Bearer ${this.accessToken}`
+        },
+        method: 'POST',
+        crossDomain: true,
+        dataType: 'json',
+        contentType: 'application/json',
+        data: JSON.stringify(payload)
+      })
+    })
+    .then((response) => {
+      const repo = {
+        fullName : response.full_name
+      };
+      context.repo = repo;
+      Object.assign(context.bindRepo, { [context.id] : repo });
+      if (context.bindBranch[context.id]) {
+        delete context.bindBranch[context.id];
+      }
+      chrome.storage.sync.set({ bindRepo: context.bindRepo });
+      return response.full_name;
+    })
+    .then(repo => {
+      return this.commitFiles(repo, 'master', [{name: "README.md", content: "initialed by gas-github"}], null, 'initial commit');
+    })
+    .then(this.getRepos.bind(this))
+    .then(updateRepo)
+    .then(updateBranch)
+    .then(() => {
+      $('#new-repo-name').val('');
+      $('#new-repo-desc').val('');
+      $('#new-repo-type').val('public');
+      showAlert(`Successfully create new repository ${repo}`);
+    })
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/leonhartX/gas-github/blob/77d86c0cbe12e28b49d988b7fdf8d52ef89bd093/src/scm/bitbucket.js#L208"></span>‘err’ is defined but never used.</td>
    </tr>
  </tbody>
</table>